php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -937,23 +1074,51 @@ private function convertJsonValueToProtoValue(
                 if (bccomp($value, "9223372036854775807") > 0) {
                     $value = bcsub($value, "18446744073709551616");
                 }
+                if (PHP_INT_SIZE == 4) {
+                    return (string) $value;
+                }
                 return $value;
             default:
                 return $value;
         }
     }
 
-    protected function mergeFromJsonArray($array)
+    /**
+     * Populates the message from a user-supplied PHP array.
+     * Array keys correspond to Message properties and nested message
+     * properties.
+     *
+     * Example:
+     * ```
+     * $message->mergeFromArray([
+     *     'name' => 'This is a message name',
+     *     'interval' => [
+     *          'startTime' => time() - 60,
+     *          'endTime' => time(),
+     *     ]
+     * ]);
+     * ```
+     *
+     * @param array $array An array containing message properties and values.
+     * @return null.
+     * @throws Exception Invalid data.
+     */
+    public function mergeFromArray($array)
```

Comment:
I don't want to make this public.

Reply:
```mergeFrom(new TestMessage([...]))``` should be ok.

Reply:
Are you saying we should allow for `Message` objects to be passed in the constructor?

Reply:
Oh I see, you want to use the existing `mergeFrom` method. Sounds good!

Reply:
I think you misunderstood my point. I don't want to change the behavior of ```mergeFrom```. I want to make ```mergeFromArray``` protected instead of public

Reply:
@TeBoring Okay, please check it now. I added it back and made it protected.